client/src/main/java/io/spine/client/ActorRequestFactory.java

```diff
@@ -57,35 +70,61 @@
     protected ActorRequestFactory(Builder builder) {
         this.actor = builder.actor;
         this.zoneOffset = builder.zoneOffset;
+        this.zoneId = builder.zoneId;
         this.tenantId = builder.tenantId;
     }
 
+    /**
+     * Creates a builder for a new request factory.
+     */
     public static Builder newBuilder() {
         return new Builder();
     }
 
+    /**
+     * Obtains the ID of the user on behalf of whom the requests are created.
+     */
     public UserId getActor() {
         return actor;
     }
 
+    /**
+     * Obtains the offset of the time zone in which the actor works.
+     */
     public ZoneOffset getZoneOffset() {
         return zoneOffset;
     }
 
+    /**
+     * Obtains the ID of the time zone in which the actor works.
+     */
+    public ZoneId getZoneId() {
+        return zoneId;
+    }
+
+    /**
+     * Obtains the ID of the tenant to which the actor belongs, or {@code null}
+     * for single-tenant execution context.
+     */
     public @Nullable TenantId getTenantId() {
         return tenantId;
     }
 
     /**
-     * Creates new factory with the same user and tenant ID, but with new time zone offset.
+     * Creates new factory with the same user and tenant ID, but with new time zone ID and offset.
+     *
+     * <p>The purpose of this method is to provide a new instance of the request factory
+     * which corresponds to new location of the actor.
      *
-     * @param zoneOffset the offset of the time zone
-     * @return new factory at new time zone
+     * @param zoneOffset the offset of the new time zone
+     * @param zoneId     the ID of the new time zone
+     * @return new factory at the new time zone
      */
-    public ActorRequestFactory switchTimezone(ZoneOffset zoneOffset) {
+    public ActorRequestFactory switchTimezone(ZoneOffset zoneOffset, ZoneId zoneId) {
```

Comment:
As discussed, we'll need to have an overloaded "shortcut" switching method, that takes `zoneId` only. The reason behind this decision is that just a `zoneId` is enough for the most cases. And those who need custom offsets, will use this one.